src/material/expansion/testing/expansion-harness.ts

```diff
@@ -0,0 +1,140 @@
+        .addOption(
+            'content', options.content,
+            (harness, content) => HarnessPredicate.stringMatches(harness.getTextContent(), content))
+        .addOption(
+            'expanded', options.expanded,
+            async (harness, expanded) => (await harness.isExpanded()) === expanded)
+        .addOption(
+            'disabled', options.disabled,
+            async (harness, disabled) => (await harness.isDisabled()) === disabled);
+  }
+
+  /** Whether the panel is expanded. */
+  async isExpanded(): Promise<boolean> {
+    return (await this.host()).hasClass('mat-expanded');
+  }
+
+  /**
+   * Gets the title text of the panel.
+   * @returns Title text or `null` if no title is set up.
+   */
+  async getTitle(): Promise<string|null> {
+    const titleEl = await this._title();
+    return titleEl ? titleEl.text() : null;
+  }
+
+  /**
+   * Gets the description text of the panel.
+   * @returns Description text or `null` if no description is set up.
+   */
+  async getDescription(): Promise<string|null> {
+    const descriptionEl = await this._description();
+    return descriptionEl ? descriptionEl.text() : null;
+  }
+
+  /** Whether the panel is disabled. */
+  async isDisabled(): Promise<boolean> {
+    return await (await this._header()).getAttribute('aria-disabled') === 'true';
+  }
+
+  /**
+   * Toggles the expanded state of the panel by clicking on the panel
+   * header. This method will not work if the panel is disabled.
+   */
+  async toggle(): Promise<void> {
+    await (await this._header()).click();
+  }
+
+  /** Gets the text content of the panel. */
+  async getTextContent(): Promise<string> {
+    return (await this._content()).text();
+  }
+
+  /**
+   * Gets a `HarnessLoader` that can be used to load harnesses for
+   * components within the panel's content area.
+   */
+  async getHarnessLoaderForContent(): Promise<HarnessLoader> {
+    return this.locatorFactory.harnessLoaderFor(EXPANSION_PANEL_CONTENT_SELECTOR);
+  }
+
+  /** Focuses the panel. */
+  async focus(): Promise<void> {
+    return (await this._header()).focus();
+  }
+
+  /** Blurs the panel. */
+  async blur(): Promise<void> {
+    return (await this._header()).blur();
+  }
+
+  /** Whether the panel has a toggle indicator displayed. */
+  async hasToggleIndicator(): Promise<boolean> {
+    return (await this._expansionIndicator()) !== null;
+  }
+
+  /** Gets the position of the toggle indicator. */
+  async getToggleIndicatorPosition(): Promise<'before'|'after'> {
+    // By default the expansion indicator will show "after" the panel header content.
+    if (await (await this._header()).hasClass('mat-expansion-toggle-indicator-before')) {
+      return 'before';
+    }
+    return 'after';
+  }
+
+  /**
+   * Checks if the given value of type `string` or `null` matches the
+   * specified exact string, pattern or `null`.
+   */
+  private static async _matchesStringNullableOption(
```

Comment:
Should we just roll support for this into `HarnessPredicate.stringMatches`? Seems like a common enough scenario

Reply:
Agreed. I moved it into `stringMatches`.